src/main/java/seedu/fridgefriend/food/Weight.java

```diff
@@ -0,0 +1,28 @@
+package seedu.fridgefriend.food;
+
+import seedu.fridgefriend.exception.InvalidQuantityException;
+
+public class Weight extends Quantity {
+
+    private String unit;
+    public static final String UNIT_CHARACTER = "g";
+
+    public Weight(String quantityString) throws InvalidQuantityException {
+        try {
+            String[] words = quantityString.split("(?<=\\d)(?=[a-z])");
+            this.number = Integer.parseInt(words[0]);
```

Comment:
Might be good to clarify what this split command does in a javadoc comment or similar.